Readme.md

```diff
@@ -3,23 +3,58 @@
 [![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T114462)
 [![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
 <!-- default badges end -->
-<!-- default file list -->
-*Files to look at*:
 
-* [Default.aspx](./CS/T114462/Default.aspx) (VB: [Default.aspx](./VB/T114462/Default.aspx))
-* [Default.aspx.cs](./CS/T114462/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/T114462/Default.aspx.vb))
-<!-- default file list end -->
-# ASPxGridView - Batch Editing - How to display save and cancel buttons only after editing
+# GridView for Web Forms - How to implement custom buttons in status bar in batch edit mode
+
 <!-- run online -->
 **[[Run Online]](https://codecentral.devexpress.com/t114462/)**
 <!-- run online end -->
 
-<p><strong>UPDATED:<br></strong><br>Starting with v18.2, if a grid contains modified data, it displays a confirmation message before a grid performs a postback or a callback. The <a href="https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.KeepChangesOnCallbacks">KeepChangesOnCallbacks</a> property specifies whether the grid supports callbacks and allows you to use the 'Preview changes' button to preview and modify inserted, deleted and edited rows before you click 'Save changes'.</p>
+In [batch edit mode](https://docs.devexpress.com/AspNet/16443/components/grid-view/concepts/edit-data/batch-edit-mode), the Grid displays three buttons in [status bar](https://docs.devexpress.com/AspNet/3693/components/grid-view/visual-elements/status-bar): _Preview changes_, _Save changes_, and _Cancel changes_. You can not change the buttons or their behavior. This example demonstrates how to replace the default buttons with custom buttons.
+
+![Grid View - Confirmation message](grid-custom-buttons.png)
+
+
+To replace the default buttons with custom buttons, implement the [StatusBar](http://docs.devexpress.devx/AspNet/DevExpress.Web.GridViewTemplates.StatusBar) template.
+
+```aspx
+<Templates>
+  <StatusBar>
+    <dx:ASPxButton ClientInstanceName="btnPreview"... />
+    <dx:ASPxButton ClientInstanceName="btnSave"... />
+    <dx:ASPxButton ClientInstanceName="btnCancel"... />
+  </StatusBar>
+</Templates>
+```
+
+Handle the [BatchEditEndEditing](http://docs.devexpress.devx/AspNet/js-ASPxClientGridView.BatchEditEndEditing) event and change the button visibility based on your conditions. In this example, custom buttons repeat the default buttons' behavior: they are hidden when the Grid has no changes and the _Show changes_ button changes its text when clicked.
```

Comment:
`In this example, custom buttons repeat the default buttons' behavior: they are hidden when the Grid has no changes and the _Show changes_ button changes its text when clicked.`

1) What do you mean by "repeat"? For example, do you mean "imitate"?

2) `the _Show changes_ button changes its text when clicked`

Do we need to mention the _new_ text to which this text changes?

Reply:
```suggestion
Handle the [BatchEditEndEditing](http://docs.devexpress.devx/AspNet/js-ASPxClientGridView.BatchEditEndEditing) event and change the button visibility based on your conditions. In this example, custom buttons imitate the default buttons' behavior: they are hidden when the Grid has no changes and the _Show changes_ button changes its text to _Hide changes_ when clicked.
```

Reply:
Thank you. I'd also like to avoid the plural possessive noun. Can we say the following?

```suggestion
Handle the [BatchEditEndEditing](http://docs.devexpress.devx/AspNet/js-ASPxClientGridView.BatchEditEndEditing) event and change the button visibility based on your conditions. In this example, custom buttons imitate the behavior of default buttons: they are hidden when the Grid has no changes and the _Show changes_ button changes its text to _Hide preview_ when clicked.
```